packages/ts-codegen/src/__tests__/context.test.ts

```diff
@@ -2,10 +2,81 @@ import ts from "typescript";
 import { parse } from "graphql";
 import { blankGraphQLTag as graphql } from "../utilities";
 import { generateTS } from "..";
-import { ContextMap } from "../context";
+import { SubTypeNamespace } from "../codegen";
+import { OutputMetadata } from "../context/utilities";
 
 describe(generateTS, () => {
   describe("Tests basic syntax GraphQL syntax", () => {
+    const contextSubTypeMetadata = {
+      managers: {
+        user: {
+          name: "user",
+          importTypeName: 'UserStateMachineType["user"]',
```

Comment:
what's the point of `importTypeName`? from the tests it seems like `UserStateMachineType` part is generated according some convention from `importPath` and `"user"` part always matches `name`. What's the point to make user type it?